apis/python/tests/test_registration_mappings.py

```diff
@@ -1217,3 +1216,105 @@ def test_append_with_nonunique_field_values(
             obs_field_name=obs_field_name,
             var_field_name=var_field_name,
         )
+
+
+# TO DO: COMMENT WHY
+@pytest.mark.parametrize("all_at_once", [False, True])
+@pytest.mark.parametrize("nobs_a", [50, 300])
+@pytest.mark.parametrize("nobs_b", [60, 400])
+def test_foo_wip(tmp_path, all_at_once, nobs_a, nobs_b):
+    obs_ids_a = [("a_%08d" % e) for e in range(nobs_a)]
+    obs_ids_b = [("b_%08d" % e) for e in range(nobs_b)]
+    var_ids = ["W", "X", "Y", "Z"]
+    obs_field_name = "cell_id"
+    var_field_name = "gene_id"
+    measurement_name = "meas"
+
+    adata = _create_anndata(
+        obs_ids=obs_ids_a,
+        var_ids=var_ids,
+        obs_field_name=obs_field_name,
+        var_field_name=var_field_name,
+        X_value_base=0,
+        measurement_name=measurement_name,
+    )
+
+    bdata = _create_anndata(
+        obs_ids=obs_ids_b,
+        var_ids=var_ids,
+        obs_field_name=obs_field_name,
+        var_field_name=var_field_name,
+        X_value_base=100,
+        measurement_name=measurement_name,
+    )
+
+    adata.obs["enum"] = pd.Categorical(obs_ids_a, categories=obs_ids_a)
+    bdata.obs["enum"] = pd.Categorical(obs_ids_b, categories=obs_ids_b)
+
+    print()
+    print(f"OK START {nobs_a} {nobs_b} {all_at_once}")
+
+    soma_uri = tmp_path.as_posix()
+
+    if all_at_once:
+        rd = tiledbsoma.io.register_anndatas(
+            None,
+            [adata, bdata],
+            measurement_name=measurement_name,
+            obs_field_name=obs_field_name,
+            var_field_name=var_field_name,
+        )
+
+        tiledbsoma.io.from_anndata(
+            soma_uri, adata, measurement_name=measurement_name, registration_mapping=rd
+        )
+        print()
+        print("OK WRITE A")
+        tiledbsoma.io.from_anndata(
+            soma_uri, bdata, measurement_name=measurement_name, registration_mapping=rd
+        )
+        print()
+        print("OK WRITE B")
+
+    else:
+        tiledbsoma.io.from_anndata(soma_uri, adata, measurement_name=measurement_name)
+        print()
+        print("OK WRITE A")
+
+        rd = tiledbsoma.io.register_anndatas(
+            soma_uri,
+            [bdata],
+            measurement_name=measurement_name,
+            obs_field_name=obs_field_name,
+            var_field_name=var_field_name,
+        )
+
+        tiledbsoma.io.from_anndata(
+            soma_uri, bdata, measurement_name=measurement_name, registration_mapping=rd
+        )
+        print()
+        print("OK WRITE B")
+
+    with tiledbsoma.Experiment.open(soma_uri) as exp:
+        obs = exp.obs.read().concat()
+        print()
+        print("OK READ OBS")
+
+        # XXX assert schema enum is categ
+        print()
+        print()
+        print("-- SOMA SCHEMA")
+        print(exp.obs.schema)
+        print("-- ARROW SCHEMA")
+        print(obs.schema)
+        print("-- PANDAS DATA")
+        print(obs.to_pandas())
+        print()
```

Comment:
Need to rm print statements from unit tests.

Reply:
Sorry @nguyenv ... `apis/python/tests/test_update_dataframes.py` was _not_ ready for review (at all) -- I shouldn't have put it up as-is ...